exporter/node_stats.py

```diff
@@ -26,70 +27,14 @@ def __init__(self, oscache, osclient):
 
     def build_cache_data(self):
         """Return list of stats to cache."""
-        nodes = self.get_nodes()
-        hosts = self.get_hosts_by_node_id()
-        reservations = self.get_reservations_by_node_id()
-
-        for node in nodes:
-            node.node_type = hosts[node.uuid]['node_type']
-            node.gpu = hosts[node.uuid]['gpu']
-            node.project_name = reservations.get(node.uuid, None)
+        nodes = node_details.get_nodes(detail=True)
+        node_details.add_node_type(nodes)
```

Comment:
Is it ever the case that you don't want node_type or project_name? I would consider putting this into `get_nodes` so that the output of `get_nodes` already has this information. Otherwise the caller needs to know that they have to call these extra functions to decorate the response.

Reply:
(And in general it keeps the API of the module small, which is usually a good thing! Makes maintenance and refactoring easier later.)

Reply:
This is true. I'll refactor